### PR TITLE
data::Solution for opm-output communication.

### PR DESCRIPTION
**Pull request by joakim-hove**
Followup to: https://github.com/OPM/opm-output/pull/101

OK - this should now be ready for review and merge.

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this with opm-output=101 opm-core=1081 please

File: opm/autodiff/GeoProps.hpp [hunk 1]
```diff
@@ -228,7 +229,7 @@ namespace Opm
         /// grid the whole TRAN keyword is quite meaningless.
 
         template <class Grid>
-        const std::vector<data::CellData> simProps( const Grid& grid ) const {
+        const CellDataContainer simProps( const Grid& grid ) const {
```
Comment (jokva) [inline opm/autodiff/GeoProps.hpp]: Regardless of name and type, there's no reason for this to be returned as `const`.

Comment (atgeirr) [inline opm/autodiff/GeoProps.hpp]: I agree, it was a small oversight. Good to correct it now.

**Comment by atgeirr** [comment]
This is in conflict with #839 which I'd like to get in first (it was delayed to get it tested with the MPI version, and a bug was discovered).

**Comment by joakim-hove** [comment]
jenkins build this with downstreams opm-common=182 opm-output=101 opm-core=1081 please

**Comment by joakim-hove** [comment]
The jenkins failure is due to a build failure in MPI code - I *assume* that is unrelated to the current changes.

**Comment by blattms** [comment]
Well, the jenkins error is unrelated and should vanish with the current master.
But the travis error might be related
```
[ 15%] Building CXX object CMakeFiles/opmsimulators.dir/opm/autodiff/NewtonIterationBlackoilCPR.cpp.o
In file included from /home/travis/build/OPM/opm-simulators/opm/autodiff/AutoDiffHelpers.hpp:26:0,
                 from /home/travis/build/OPM/opm-simulators/opm/autodiff/NewtonIterationBlackoilCPR.cpp:29:
/home/travis/build/OPM/opm-simulators/opm/autodiff/GeoProps.hpp:36:37: fatal error: opm/output/data/Cells.hpp: No such file or directory
 #include <opm/output/data/Cells.hpp>
                                     ^
compilation terminated.
make[2]: *** [CMakeFiles/opmsimulators.dir/opm/autodiff/NewtonIterationBlackoilCPR.cpp.o] Error 1
make[1]: *** [CMakeFiles/opmsimulators.dir/all] Error 2
make: *** [all] Error 2

**Comment by joakim-hove** [comment]
The travis error is certainly related, but that should be fixed in the other PRs.

**Comment by joakim-hove** [comment]
jenkins build this with downstreams opm-common=182 opm-output=101 opm-core=1081 please

**Comment by joakim-hove** [comment]
OK - this has been rebased and should be ready for review.

**Comment by joakim-hove** [comment]
jenkins build this with opm-common=182 opm-output=101 opm-core=1081 please

**Comment by joakim-hove** [comment]
jenkins build this with opm-output=101 opm-core=1081 please

File: opm/autodiff/Compat.hpp [hunk 2]
```diff
@@ -69,13 +70,11 @@ inline std::vector< double >& stripe( const std::vector< double >& v,
 
 
 
-inline data::Solution simToSolution( const SimulationDataContainer& reservoir,
-                             PhaseUsage phases ) {
-    using ds = data::Solution::key;
-
+    inline data::Solution simToSolution( const SimulationDataContainer& reservoir,
```
Comment (atgeirr) [inline opm/autodiff/Compat.hpp]: Indent?

**Comment by atgeirr** [comment]
This review was much quicker than I anticipated. It looks good, feel free to merge at will.

**Pull request merged**